stackify/transport/agent/agent_socket.py

```diff
@@ -0,0 +1,38 @@
+import logging
+import os
+import retrying
+import requests_unixsocket
+
+
+internal_logger = logging.getLogger(__name__)
+
+
+class AgentSocket(object):
+    SOCKET_LOG_FILE = 'http+unix://%2Fusr%2Flocal%2Fstackify%2Fstackify.sock'
+    SOCKET_SCHEME = 'http+unix://'
+
+    def __init__(self):
+        self._session = requests_unixsocket.Session()
+
+    def _post(self, url, payload):
+        if not url.startswith(self.SOCKET_SCHEME):
+            url = os.path.join(self.SOCKET_LOG_FILE, url.lstrip('/'))
+
+        internal_logger.debug('Request URL: {}'.format(url))
+        internal_logger.debug('POST data: {}'.format(payload))
+
+        headers = {
+            'Content-Type': 'application/x-protobuf',
+        }
+
+        try:
+            response = self._session.post(url, payload, headers=headers)
+            internal_logger.debug('Response status: {}'.format(response.status_code))
+            return response
+        except Exception as e:
+            internal_logger.debug('HTTP UNIX Socket domain exception: {}.'.format(e))
+            raise
+
+    @retrying.retry(wait_exponential_multiplier=1000, stop_max_delay=10000)
```

Comment:
can we set a max amount of retries to 6 as well.  Are the time values here millis ?

Reply:
does the stop_max_delay indicate the max amount of delay and also stop retrying after that is hit ?  In that situation, this would be 10 - and that is fine.

Reply:
yes, the numbers are in millis.
but this is an exponential backoff which means the retry delay is different in each retry.
and it will increase on each retry.
seems like this will only take 4 or 5 retry.
1st retry will be done after 1 second delay
2nd retry will be done after 2 seconds delay
3rd retry will be done after 4 seconds delay
4th retry wlll be done after 8 seconds delay
and I guess the 5th retry will wait 10 seconds delay since that is the max value specified.

which means it will take 25 seconds if all requests failed.
compared to 10 retry in just a span of 10 seconds.

Reply:
will it continue retrying after the 5th retry of 10 seconds?

Reply:
currently doing some tests.. and the current implementation will only do 4 requests where 3 are retries.
the first retry seems like waiting for 2 seconds.
second retry will wait for 4 seconds
and third retry will wait for 8 seconds.
since the max waiting time is 10 seconds and the next wait is 16 seconds it will not do a retry after the third one.

Reply:
updating the max delay to 32 seconds so it will do 5 retries..
6 request all in all including the initial request.
total delay if all requests fail will be 64 seconds.